### ISSUE TITLE
Hangfire failed to start after stopping ,String or binary data would be truncated.

### ISSUE THREAD

**Issue by jason840917gmail** [HangfireIO/Hangfire#2173]
HangFire version: 1.7.32

**Stopping Process**

```
`3/1/2023 6:51:53 PM (4248) [Hangfire.AppBuilderExtensions] Web application is shutting down via OWIN's host.OnAppDisposing callback.
3/1/2023 6:51:53 PM (4248) [Hangfire.Server.BackgroundServerProcess] Server s096v159:1812:45675f88 caught stopping signal...
3/1/2023 6:51:53 PM (7968) [Hangfire.Server.BackgroundServerProcess] Server s096v159:1812:45675f88 caught stopped signal...
3/1/2023 6:51:54 PM (10488) [Hangfire.Server.BackgroundServerProcess] Server s096v159:1812:45675f88 All dispatchers stopped
3/1/2023 6:51:54 PM (10488) [Hangfire.Server.BackgroundServerProcess] Server s096v159:1812:45675f88 successfully reported itself as stopped in 6.5134 ms
3/1/2023 6:51:54 PM (10488) [Hangfire.Server.BackgroundServerProcess] Server s096v159:1812:45675f88 has been stopped in total 889.7479 ms
3/1/2023 6:51:54 PM (4248) [Hangfire.Server.AspNetShutdownDetector] ASP.NET application is shutting down: HostingEnvironment.StopListening.


```
**After stopping the server failed to start again with the fallowing error:**

```
3/2/2023 5:45:27 AM (10772) [Hangfire.BackgroundJobServer] Starting Hangfire Server using job storage: 'SQL Server (custom)'
3/2/2023 5:45:27 AM (10772) [Hangfire.BackgroundJobServer] Using the following options for SQL Server job storage: Queue poll interval: 00:00:00.
3/2/2023 5:45:27 AM (10772) [Hangfire.BackgroundJobServer] Using the following options for Hangfire Server:
    Worker count: 20
    Listening queues: 'default'
    Shutdown timeout: 00:00:15
    Schedule polling interval: 00:00:15
3/2/2023 5:45:28 AM (12048) [Hangfire.Processing.BackgroundExecution] Execution BackgroundServerProcess is in the Failed state now due to an exception, execution will be retried no more than in 00:00:15
Microsoft.Data.SqlClient.SqlException (0x80131904): String or binary data would be truncated.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction) in D:\a\_work\1\s\src\Microsoft.Data.SqlClient\netfx\src\Microsoft\Data\SqlClient\SqlConnection.cs:line 2363
   at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction) in D:\a\_work\1\s\src\Microsoft.Data.SqlClient\src\Microsoft\Data\SqlClient\SqlInternalConnection.cs:line 770
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose) in D:\a\_work\1\s\src\Microsoft.Data.SqlClient\netfx\src\Microsoft\Data\SqlClient\TdsParser.cs:line 1835
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady) in D:\a\_work\1\s\src\Microsoft.Data.SqlClient\netfx\src\Microsoft\Data\SqlClient\TdsParser.cs:line 3067
   at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted) in D:\a\_work\1\s\src\Microsoft.Data.SqlClient\netfx\src\Microsoft\Data\SqlClient\SqlCommand.cs:line 5729
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest) in D:\a\_work\1\s\src\Microsoft.Data.SqlClient\netfx\src\Microsoft\Data\SqlClient\SqlCommand.cs:line 5555
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry) in D:\a\_work\1\s\src\Microsoft.Data.SqlClient\netfx\src\Microsoft\Data\SqlClient\SqlCommand.cs:line 5159
   at Microsoft.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry) in D:\a\_work\1\s\src\Microsoft.Data.SqlClient\netfx\src\Microsoft\Data\SqlClient\SqlCommand.cs:line 2032
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteNonQuery() in D:\a\_work\1\s\src\Microsoft.Data.SqlClient\netfx\src\Microsoft\Data\SqlClient\SqlCommand.cs:line 1475
   at Dapper.SqlMapper.ExecuteCommand(IDbConnection cnn, CommandDefinition& command, Action`2 paramReader)
   at Dapper.SqlMapper.ExecuteImpl(IDbConnection cnn, CommandDefinition& command)
   at Hangfire.SqlServer.SqlServerConnection.<>c__DisplayClass17_0.<AnnounceServer>b__0(DbConnection connection)
   at Hangfire.SqlServer.SqlServerStorage.<>c__DisplayClass35_0.<UseConnection>b__0(DbConnection connection)
   at Hangfire.SqlServer.SqlServerStorage.UseConnection[T](DbConnection dedicatedConnection, Func`2 func)
   at Hangfire.SqlServer.SqlServerStorage.UseConnection(DbConnection dedicatedConnection, Action`1 action)
   at Hangfire.SqlServer.SqlServerConnection.AnnounceServer(String serverId, ServerContext context)
   at Hangfire.Server.BackgroundServerProcess.CreateServer(BackgroundServerContext context)
   at Hangfire.Server.BackgroundServerProcess.Execute(Guid executionId, BackgroundExecution execution, CancellationToken stoppingToken, CancellationToken stoppedToken, CancellationToken shutdownToken)
   at Hangfire.Server.BackgroundProcessingServer.RunServer(Guid executionId, Object state)
   at Hangfire.Processing.BackgroundExecution.Run(Action`2 callback, Object state)
ClientConnectionId:7d2a6a7d-db2f-4c58-a59e-3365e79f1961
Error Number:8152,State:30,Class:16`
```

**Comment by odinserj** [comment]
Will be released with 1.8.0, pre-release version [1.8.0-rc3](https://www.hangfire.io/blog/2023/03/30/hangfire-1.8.0-rc3.html) already available that contains a migration for this.

**Comment by odinserj** [comment]
Great @jason840917gmail, thank you for your patience 🙏
